Frends.SFTP.UploadFiles/Frends.SFTP.UploadFiles/Definitions/FileTransporter.cs

```diff
@@ -361,17 +373,15 @@ private static Encoding GetEncoding(Destination dest)
             case FileEncoding.UTF8:
                 return dest.EnableBomForFileName ? new UTF8Encoding(true) : new UTF8Encoding(false);
             case FileEncoding.ASCII:
-                return Encoding.ASCII;
+                return new ASCIIEncoding();
```

Comment:
Actually this was mistake. The method in FileTransporter should be Encoding for file name and the one in SingleFileTransfer is for file content. Fixed.

Reply:
Sorry I was not very clear on the this thing. Essentially we want to have those methods not duplicated, but as one method that we can call from multiple places. Because any bugfixes or related things then can be handled by fixing one place instead of having to remember multiple ones. For you as the code's author this may be simple to remember and to know beforehand, but for someone not familiar with the codebase it is exponentially harder to know that there are actually two places that handle encoding. The two current methods are essentially identical (as they should be) and between those we do not want the implementation to be different at any point in time. Thus if you just extract the method e.g. like this:

```
private static Encoding GetEncoding(Enum wellKnownEncoding, string customEncoding, bool enableUtfBom)
{
    switch (wellKnownEncoding)
    {
        case FileEncoding.UTF8:
            return enableUtfBom ? new UTF8Encoding(true) : new UTF8Encoding(false);
        case FileEncoding.ASCII:
            return new ASCIIEncoding();
        case FileEncoding.ANSI:
            return Encoding.Default;
        case FileEncoding.WINDOWS1252:
            return CodePagesEncodingProvider.Instance.GetEncoding("windows-1252");
        case FileEncoding.Other:
            return CodePagesEncodingProvider.Instance.GetEncoding(customEncoding);
        default:
            throw new ArgumentOutOfRangeException($"Unknown Encoding type: '{encoding}'.");
    }
}
```

So we get rid of the `Destination` class dependency in the method parameter and thus we just introduce only the parameters that are actually needed for the method to work. And now you can call the method like `__helpers__.GetEncoding(dest.FileNameEncoding, dest.FileNameEncodingInString, dest.EnableBomForFileName)`.

This makes sure that 
1. Anyone fixing anything encoding related will not need to know about all the implementation places
2. Anyone reading the code will know exactly what parameters does the `GetEncoding` need (instead of guessing what part of the `Destination` class is actually used).
3. I won't bug you with this place anymore :D

Reply:
Now I understood what you meant. I've made the necessary changes and now the method is called from Util class.